adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala

```diff
@@ -578,9 +590,35 @@ class ADAMContext(val sc: SparkContext) extends Serializable with Logging {
     }
   }
 
+  def loadPairedFastQ(filePath1: String,
+                      filePath2: String,
+                      stringency: ValidationStringency,
+                      recordGroupOpt: Option[String]): RDD[(AlignmentRecord, AlignmentRecord)] = {
+    val reads1 = loadUnpairedFastq(filePath1, recordGroupOpt, setFirstOfPair = true)
+    val reads2 = loadUnpairedFastq(filePath2, recordGroupOpt, setSecondOfPair = true)
+
+    stringency match {
+      case ValidationStringency.STRICT | ValidationStringency.LENIENT =>
+        val count1 = reads1.count
+        val count2 = reads2.count
+
+        if (count1 != count2) {
+          val msg = s"Fastq 1 ($filePath1) has $count1 reads, fastq 2 ($filePath2) has $count2 reads"
+          if (stringency == ValidationStringency.STRICT)
+            throw new IllegalArgumentException(msg)
+          else if (stringency == ValidationStringency.LENIENT)
+            logError(msg)
```

Comment:
I suppose you don't _need to_, but doing the join would guarantee correct output. If I was calling this function with Lenient validation stringency and mismatched reads, that is the semantics I would expect. What are your thoughts? Does my view of the semantics seem reasonable or unreasonable?